controllers/hetznercluster_controller_test.go

```diff
@@ -844,55 +843,71 @@ var _ = Describe("reconcileRateLimit", func() {
-		}
-		if (hetznerCluster.Status.ControlPlaneLoadBalancer.IPv4 != "<nil>"){
-			t.Fatalf("Wrong value for Status.ControlPlaneLoadBalancer.IPv4. Got: %s, Want: '<nil>'", hetznerCluster.Status.ControlPlaneLoadBalancer.IPv4)
+		if got != true {
+			t.Fatalf("return value should be true")
 		}
-		assert.Equal(t, false, got)
 	})
-	t.Run("function should return true if hetznerCluster.Spec.ControlPlaneLoadBalancer.Enabled is true and hetznerCluster.Status.ControlPlaneLoadBalancer.IPv4 contains a value (not nil)", func(t *testing.T) {
-		hetznerCluster := &infrav1.HetznerCluster{Spec: infrav1.HetznerClusterSpec{ControlPlaneLoadBalancer: infrav1.LoadBalancerSpec{Enabled: true, Port: 12}}, Status: infrav1.HetznerClusterStatus{ControlPlaneLoadBalancer: &infrav1.LoadBalancerStatus{IPv4: "xyz"}}}
+
+	t.Run("return false if load balancer is enabled but IPv4 is '<nil>'", func(t *testing.T) {
+		hetznerCluster := &infrav1.HetznerCluster{
+			Spec: infrav1.HetznerClusterSpec{
+				ControlPlaneLoadBalancer: infrav1.LoadBalancerSpec{Enabled: true}, 
+				ControlPlaneEndpoint: nil}, 
+				Status: infrav1.HetznerClusterStatus{
+					ControlPlaneLoadBalancer: &infrav1.LoadBalancerStatus{
+						IPv4: "<nil>"},
+				},
+		}
 		got := SetControlPlaneEndpoint(hetznerCluster)
-		if (hetznerCluster.Status.ControlPlaneLoadBalancer.IPv4 != "xyz"){
-			t.Fatalf("Wrong input for hetznerCluster.Status.ControlPlaneLoadBalancer.IPv4. Got: %s, Want: 'xyz'", hetznerCluster.Status.ControlPlaneLoadBalancer.IPv4)
+		// ControlPlaneEnpdoint should still remain nil
+		if (hetznerCluster.Spec.ControlPlaneEndpoint != nil){
+			t.Fatalf("ControlPlaneEndpoint should not change. It should remain nil")
 		}
-		if (hetznerCluster.Spec.ControlPlaneLoadBalancer.Port != 12){
-			t.Fatalf("Wrong input for hetznerCluster.Spec.ControlPlaneLoadBalancer.Port. Got: %d, Want: 12", hetznerCluster.Spec.ControlPlaneLoadBalancer.Port)
+		if got != false {
+			t.Fatalf("return value should be false")
 		}
-		assert.Equal(t, true, got)
 	})
-	t.Run("function should return true if hetznerCluster.Spec.ControlPlaneLoadBalancer.Enabled is true, hetznerCluster.Status.ControlPlaneLoadBalancer.IPv4 is not nil and hetznerCluster.Spec.ControlPlaneEndpoint is nil", func(t *testing.T) {
-		hetznerCluster := &infrav1.HetznerCluster{Spec: infrav1.HetznerClusterSpec{ControlPlaneLoadBalancer: infrav1.LoadBalancerSpec{Enabled: true, Port: 11}, ControlPlaneEndpoint: nil}, Status: infrav1.HetznerClusterStatus{ControlPlaneLoadBalancer: &infrav1.LoadBalancerStatus{IPv4: "xyz"}}}
+
+	t.Run("return true if load balancer is enabled and IPv4 is not nil, but ControlPlaneEndpoint is nil. Values of ControlPlaneEndpoint.Host and ControlPlaneEndpoint.Port will get updated", func(t *testing.T) {
+		hetznerCluster := &infrav1.HetznerCluster{
+			Spec: infrav1.HetznerClusterSpec{
+				ControlPlaneLoadBalancer: infrav1.LoadBalancerSpec{
+					Enabled: true, 
+					Port: 11}, 
```

Comment:
always fix the line break. you never do it this way with structs, you always close it in the next line, i.e.

Port: 11,
},